src/lib/Runtime.hs

```diff
@@ -160,3 +160,16 @@ readStream h action = go
     go = do
       eof <- hIsEOF h
       unless eof $ hGetLine h >>= action >> go
+
+-- ==== thread-local storage ===
+
+-- We use TLS to implement some dynamically scoped variables, like the file
+-- descriptor for the current output stream. We allocate the thread-local memory
+-- here, and we generate code for `pthread_setspecific` at entry functions and
+-- `pthread_getspecific` at use sites.
+
+foreign import ccall "dex_pthread_key_create"
+  pthreadKeyCreate :: IO (Ptr ())
+
+allocateTLS :: IO (Ptr (Ptr ()))
+allocateTLS = castPtr <$> pthreadKeyCreate
```

Comment:
This is a bad name, since it doesn't actually allocate the TLS. It only allocates a key that lets you reach the TLS.